canonical/cloud-init
# Update WebHookHandler to run as background thread (SC-456)

## Proposed Commit Message


```
Update WebHookHandler to run as background thread

When reporting via the WebHookHandler, the events are handled on the
main thread. This slows the execution of cloud-init significantly.
If there are timeouts in POSTing, this can make cloud-init appear to be
stuck.

This commit updates the WebHookHandler to run in the background as to
not block the main thread.

Documentation was also added as this feature wasn't previously
documented outside of examples.

LP: #1910552
```

## Additional Context
https://bugs.launchpad.net/maas/+bug/1910552

I intentionally didn't document the Azure-specific reporter as I'm not sure the current state of it or if they would want it enabled arbitrarily. I plan on discussing it at our next Azure sync.

## Test Steps
I used `docker run -p 55555:55555 darklynx/request-baskets` on my host machine to ensure requests are coming through or canceled as expected. 

Note that I moved two unit test files into a subdirectory for organization. No changes to those two files other than the move, and I added a 3rd file for testing this behavior.

## Checklist:
